volatility/framework/plugins/windows/vadinfo.py

```diff
@@ -102,17 +108,31 @@ def list_vads(cls, proc: interfaces.objects.ObjectInterface,
 
     @classmethod
     def vad_dump(cls, context: interfaces.context.ContextInterface, proc: interfaces.objects.ObjectInterface,
-                 vad: interfaces.objects.ObjectInterface) -> interfaces.plugins.FileInterface:
+                 vad: interfaces.objects.ObjectInterface, 
+                 maxsize = MAXSIZE_DEFAULT) -> interfaces.plugins.FileInterface:
         """Extracts the complete data for Vad as a FileInterface.
 
         Args:
             context: The context to retrieve required elements (layers, symbol tables) from
             proc: an _EPROCESS instance
             vad: The suspected VAD to extract (ObjectInterface)
+            maxsize: Max size of VAD section (default MAXSIZE_DEFAULT)
 
         Returns:
             A FileInterface object containing the complete data for the process or None in the case of failure
         """
+
+        try:
+            vad_start = vad.get_start()
+            vad_end = vad.get_end()
+        except Exception as excp:
```

Comment:
Ideally is a very generic exception, and it would be much better to catch just the expected exceptions (so we don't catch things like KeyboardInterrupt, etc).